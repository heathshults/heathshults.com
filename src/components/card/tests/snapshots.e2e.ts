import { snapIt } from '../../../test';

const component = 'card';

describe(component, () => {
  const snap = snapIt(component);

  describe('renders', () => {
    snap('a card', '<hs-card></hs-card>');
    snap(
      'all the child elements correctly',
      `<hs-card>
      <hs-image></hs-image>
      <hs-card-header>
        <h2 class="hs-heading u-xlarge">
          Heading
          <div class="hs-heading__sub">Sub-heading</div>
        </h2>
      </hs-card-header>
      <hs-card-body>
        <p class="hs-paragraph">
          Lorem ipsum dolor sit amet, feugiat corpora ex eam. Inciderint eloquentiam sea et.
        </p>
      </hs-card-body>
      <hs-card-footer>
        <div class="hs-input-group">
          <button class="hs-button hs-button--block hs-button--brand">Button</button>
          <button class="hs-button hs-button--block hs-button--info">Button</button>
        </div>
      </hs-card-footer>
    </hs-card>`
    );
    snap(
      'all the child elements correctly',
      `<hs-card>
      <hs-media-item>
        <hs-media-image src="https://placehold.it/80"></hs-media-image>
        <hs-media-body>
          <h2 class="hs-heading">Title
            <span class="hs-heading__sub">Subtitle</span>
          </h2>
          <p class="hs-paragraph">
            Lorem ipsum dolor sit amet, feugiat corpora ex eam. Lorem ipsum dolor sit amet, feugiat corpora ex eam. Lorem ipsum dolor
            sit amet, feugiat corpora ex eam.
          </p>
        </hs-media-body>
      </hs-media-item>
    </hs-card>`
    );
  });
});
