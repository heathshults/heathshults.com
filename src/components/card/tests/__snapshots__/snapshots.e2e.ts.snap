// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card renders a card 1`] = `"<hs-card class=\\"hydrated\\"><div class=\\"hs-card\\"></div></hs-card>"`;

exports[`card renders all the child elements correctly 1`] = `
"<hs-card class=\\"hydrated\\"><div class=\\"hs-card\\"><hs-image class=\\"hydrated\\"><img class=\\"hs-image\\" src=\\"https://source.unsplash.com/random/\\"></hs-image><hs-card-header class=\\"hydrated\\"><header class=\\"hs-card__header\\"><h2 class=\\"hs-heading u-xlarge\\">
          Heading
          <div class=\\"hs-heading__sub\\">Sub-heading</div></h2></header></hs-card-header><hs-card-body class=\\"hydrated\\"><div class=\\"hs-card__body\\"><p class=\\"hs-paragraph\\">
          Lorem ipsum dolor sit amet, feugiat corpora ex eam. Inciderint eloquentiam sea et.
        </p></div></hs-card-body><hs-card-footer class=\\"hydrated\\"><footer class=\\"hs-card__footer\\"><div class=\\"hs-input-group\\"><button class=\\"hs-button hs-button--block hs-button--brand\\">Button</button><button class=\\"hs-button hs-button--block hs-button--info\\">Button</button></div></footer></hs-card-footer></div></hs-card>"
`;

exports[`card renders all the child elements correctly 2`] = `
"<hs-card class=\\"hydrated\\"><div class=\\"hs-card\\"><hs-media-item class=\\"hydrated\\"><div class=\\"hs-card__item hs-media\\"><hs-media-image src=\\"https://placehold.it/80\\" class=\\"hydrated\\"><div class=\\"hs-media__image\\"><img class=\\"hs-image\\" src=\\"https://placehold.it/80\\"></div></hs-media-image><hs-media-body class=\\"hydrated\\"><div class=\\"hs-media__body\\"><h2 class=\\"hs-heading\\">Title
            <span class=\\"hs-heading__sub\\">Subtitle</span></h2><p class=\\"hs-paragraph\\">
            Lorem ipsum dolor sit amet, feugiat corpora ex eam. Lorem ipsum dolor sit amet, feugiat corpora ex eam. Lorem ipsum dolor
            sit amet, feugiat corpora ex eam.
          </p></div></hs-media-body></div></hs-media-item></div></hs-card>"
`;
